plugins/search/src/apis.ts

```diff
@@ -35,12 +37,19 @@ class SearchApi {
 
   private async entities() {
     const entities = await this.catalogApi.getEntities();
-    return entities.items.map((result: any) => ({
-      name: result.metadata.name,
-      description: result.metadata.description,
-      owner: result.spec.owner,
-      kind: result.kind,
-      lifecycle: result.spec.lifecycle,
+    return entities.items.map((entity: Entity) => ({
+      name: entity.metadata.name,
+      description: entity.metadata.description || 'No description',
+      owner:
+        typeof entity.spec?.owner === 'string' ? entity.spec?.owner : 'unknown',
```

Comment:
In general when making a fallback like this, an idiomatic way is to use `??` when you know that it's always an undefined that you are replacing, or the good old `||` if you want to be liberal in replacing everything falsy.

Here you know that the type is always going to be a string, so what you actually wanted was probably the former, but could also use the latter to cover empty strings (which should be forbidden, but ...).

TL:DR....: `owner: entity.spec?.owner ?? 'unknown'` probably fits well here, to avoid repetition.

Reply:
Before I had empty columns as you mentioned below, but the problem I experienced here was that I'm seeing owners can be "string | number | boolean | JsonObject | JsonArray" 🤔  which was the reason I had to check that it was a string here

Reply:
It might be something I do wrong with types that can be solved instead

Reply:
Ohh I see, it's because you're looking at the generic Entity type and it says that "in general, a spec can contain anything that json accepts", while the actual subtypes narrow it down to "these actual fields are the ones that actually exist".

Yeah OK. So `typeof entity.spec?.owner === 'string' ? entity.spec.owner : undefined` could perhaps make sense? The alternative would be to (perhaps in a separate function) switch on the `kind`, cast to the right type for that, and read. But that may not buy us much here.